src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1172,11 +1170,9 @@ static bool processContextAttributeVector (ContextElement*
     for (unsigned int ix = 0; ix < ceP->contextAttributeVector.size(); ++ix) {
 
         ContextAttribute* targetAttr = ceP->contextAttributeVector.get(ix);
-
         /* No matter if success or fail, we have to include the attribute in the response */
         ContextAttribute* ca = new ContextAttribute(targetAttr->name, targetAttr->type);
-        setResponseMetadata(targetAttr, ca);
-        cerP->contextElement.contextAttributeVector.push_back(ca);
+        setResponseMetadata(targetAttr, ca);        
 
```

Comment:
It seems to me you could do

    cerP->contextElement.contextAttributeVector.push_back(ca);

right here, instead of in eleven different locations.
I might be wrong ...